oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/logout/OidcEndpointLogoutHandler.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.client.oidc.logout;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+import org.springframework.util.Assert;
+import org.springframework.util.ResourceUtils;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A logout handler for coordinating logout with an OIDC OP on the server side.
+ *
+ * The use of the end_session_endpoint as a server-side endpoint isn't specified; however,
+ * there are OIDC OPs that support it.
+ *
+ * @author Josh Cummings
+ * @since 5.1
+ * @see <a href="http://openid.net/specs/openid-connect-session-1_0.html#RPLogout">RP-Initiated Logout</a>
+ */
+public class OidcEndpointLogoutHandler implements LogoutHandler {
+	private final Logger log = LoggerFactory.getLogger(this.getClass());
+
+	private final RestTemplate restTemplate;
+
+	private boolean includeIdTokenHint;
+
+	private String oidcLogoutEndpointUrl;
```

Comment:
This needs to pull configuration from client registrations as opposed to being hardcoded. I still need to take a look at ClientRegistrationRepository.

Reply:
Also, note that there is actually no place yet via client registration to specify this endpoint.